numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -729,6 +731,23 @@ VOID_getitem(void *input, void *vap)
         PyArray_UpdateFlags((PyArrayObject *)ret, NPY_ARRAY_UPDATE_ALL);
         return (PyObject *)ret;
     }
+    
+    if (!future_warning_emitted){
+        /* 2017-11-17, 1.14 */
+        if (DEPRECATE_FUTUREWARNING(
+                "the `.item()` method of unstructured void types will return "
+                "an immutable `bytes` object in the near future instead of the "
+                "mutable memoryview or integer array returned in numpy 1.13.")
```

Comment:
Maybe worth noting that this is already possible with `bytes(void_obj)` and/or `bytearray(void_obj)`